virttest/qemu_vm.py

```diff
@@ -4224,4 +4224,18 @@ def balloon(self, size):
             size = "%s MB" % size
         normalize_data_size = utils_misc.normalize_data_size
         size = int(float(normalize_data_size(size, 'B', '1024')))
-        return self.monitor.balloon(size)
+        # Clear the event list of all QMP monitors
```

Comment:
IMO,  these new codes should be added in  class QMPMonitor().balloon().
and if so, it's no need to check the vm.protocol.